packages/neutrino/index.js

```diff
@@ -8,10 +8,15 @@ const configPrefix = 'neutrino.config';
 module.exports = (middleware = { use: ['.neutrinorc.js'] }, options = {}) => {
   const neutrino = new Neutrino(options);
   const { argv } = yargs;
-  const mode = argv.mode || 'production';
+  let { mode } = argv;
 
-  if (!process.env.NODE_ENV) {
+  if (mode) {
     process.env.NODE_ENV = mode;
```

Comment:
Here we force `NODE_ENV` to always be the same value as `--mode`, but that's not what we want right? What about `NODE_ENV=development webpack --mode production`?

Reply:
This was intentional (there's a note in the PR description), for the reasons under scenario G here:
https://github.com/mozilla-neutrino/neutrino-dev/pull/955#issuecomment-399957959

...though open to counter-arguments :-)